gensrc/script/functions.py

```diff
@@ -692,18 +692,21 @@
-    [150093, 'array_distinct', 'ARRAY_INT',       ['ARRAY_INT'],       'ArrayFunctions::array_distinct_int'],
-    [150094, 'array_distinct', 'ARRAY_BIGINT',    ['ARRAY_BIGINT'],    'ArrayFunctions::array_distinct_bigint'],
-    [150095, 'array_distinct', 'ARRAY_LARGEINT',  ['ARRAY_LARGEINT'],  'ArrayFunctions::array_distinct_largeint'],
-    [150096, 'array_distinct', 'ARRAY_FLOAT',     ['ARRAY_FLOAT'],     'ArrayFunctions::array_distinct_float'],
-    [150097, 'array_distinct', 'ARRAY_DOUBLE',    ['ARRAY_DOUBLE'],    'ArrayFunctions::array_distinct_double'],
-    [150098, 'array_distinct', 'ARRAY_VARCHAR',   ['ARRAY_VARCHAR'],   'ArrayFunctions::array_distinct_varchar'],
-    [150099, 'array_distinct', 'ARRAY_DECIMALV2', ['ARRAY_DECIMALV2'], 'ArrayFunctions::array_distinct_decimalv2'],
-    [150100, 'array_distinct', 'ARRAY_DATETIME',  ['ARRAY_DATETIME'],  'ArrayFunctions::array_distinct_datetime'],
-    [150101, 'array_distinct', 'ARRAY_DATE',      ['ARRAY_DATE'],      'ArrayFunctions::array_distinct_date'],
+    [150090, 'array_distinct', 'ARRAY_BOOLEAN',     ['ARRAY_BOOLEAN'],     'ArrayFunctions::array_distinct<TYPE_BOOLEAN>'],
+    [150091, 'array_distinct', 'ARRAY_TINYINT',     ['ARRAY_TINYINT'],     'ArrayFunctions::array_distinct<TYPE_TINYINT>'],
+    [150092, 'array_distinct', 'ARRAY_SMALLINT',    ['ARRAY_SMALLINT'],    'ArrayFunctions::array_distinct<TYPE_SMALLINT>'],
+    [150093, 'array_distinct', 'ARRAY_INT',         ['ARRAY_INT'],         'ArrayFunctions::array_distinct<TYPE_INT>'],
+    [150094, 'array_distinct', 'ARRAY_BIGINT',      ['ARRAY_BIGINT'],      'ArrayFunctions::array_distinct<TYPE_BIGINT>'],
+    [150095, 'array_distinct', 'ARRAY_LARGEINT',    ['ARRAY_LARGEINT'],    'ArrayFunctions::array_distinct<TYPE_LARGEINT>'],
+    [150096, 'array_distinct', 'ARRAY_FLOAT',       ['ARRAY_FLOAT'],       'ArrayFunctions::array_distinct<TYPE_FLOAT>'],
+    [150097, 'array_distinct', 'ARRAY_DOUBLE',      ['ARRAY_DOUBLE'],      'ArrayFunctions::array_distinct<TYPE_DOUBLE>'],
+    [150098, 'array_distinct', 'ARRAY_VARCHAR',     ['ARRAY_VARCHAR'],     'ArrayFunctions::array_distinct<TYPE_VARCHAR>'],
+    [150099, 'array_distinct', 'ARRAY_DECIMALV2',   ['ARRAY_DECIMALV2'],   'ArrayFunctions::array_distinct<TYPE_DECIMALV2>'],
+    [150100, 'array_distinct', 'ARRAY_DATETIME',    ['ARRAY_DATETIME'],    'ArrayFunctions::array_distinct<TYPE_DATETIME>'],
+    [150101, 'array_distinct', 'ARRAY_DATE',        ['ARRAY_DATE'],        'ArrayFunctions::array_distinct<TYPE_DATE>'],
+    [150102, 'array_distinct', 'ARRAY_DECIMAL32',   ['ARRAY_DECIMAL32'],   'ArrayFunctions::array_distinct<TYPE_DECIMAL32>'],
+    [150103, 'array_distinct', 'ARRAY_DECIMAL64',   ['ARRAY_DECIMAL64'],   'ArrayFunctions::array_distinct<TYPE_DECIMAL64>'],
+    [150104, 'array_distinct', 'ARRAY_DECIMAL128',  ['ARRAY_DECIMAL128'],  'ArrayFunctions::array_distinct<TYPE_DECIMAL128>'],
```

Comment:
How about support the `array_distinct` for JSON type?

Reply:
Since JSON has already support the compare operation

Reply:
support next pr